src/main/java/de/leonhard/storage/Config.java

```diff
@@ -56,6 +58,34 @@ public Config(final File file) {
     super(file);
   }
 
+  public void annotateClass(Object classInstance) {
+    Class<?> clazz = classInstance.getClass();
+    try {
+      for (Field field : clazz.getFields()) {
+        if(field.isAnnotationPresent(ConfigPath.class)) {
+          field.setAccessible(true);
+          field.set(classInstance, this.get(field.getAnnotation(ConfigPath.class).value(), field.getType()));
+        }
+      }
+    }catch (IllegalAccessException e) {
+      e.printStackTrace();
+    }
+  }
+
+  public void annotateClass(Object classInstance, String section) {
```

Comment:
I have an Idea to reduce code duplications:

You have:
```java
public void annotateClass(Object classInstance) {...}
public void annotateClass(Object classInstance, String section) {...}
```

And both just call:
```java
public void annotateClass(Object classInstance, UnaryOperator<String> elementSelector) {...}
```
You would pass the `configPath.value()` as parameter to the `elementSelector` and maybe you could even use a `BiFunction<String, Field, String>` so you could react on the field and so?

Reply:
This is just an Idea to future proof this Idea.

Reply:
I think this is what you meant
![image](https://user-images.githubusercontent.com/62883418/145706351-d156080c-ec14-47ab-b5a5-d8bdd8f283fe.png)

Reply:
Yes that is what I meant, but maybe using a BiFunction<String, Field, String> would be even more elegant.